keyboards/converter/usb_usb/keymaps/chriskopher/tap_dance.c

```diff
@@ -0,0 +1,180 @@
+/* Copyright 2020 Christopher Ko <chriskopher@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "config.h"
+#include "shared_enum.h"
+
+#include "quantum.h"
+#include "action.h"
+#include "action_code.h"
+#include "process_tap_dance.h"
+
+enum tap_dance_state {
+    SINGLE_HOLD = 1,
+    SINGLE_TAP,
+    DOUBLE_TAP,
+    TRIPLE_TAP,
+};
+
+// Record current state of a tap dance
+static int espc_l_tap_state   = 0;
+static int espc_r_tap_state   = 0;
+static int scln_cln_tap_state = 0;
+
+// Watch the state of the tap dance
+int cur_dance(qk_tap_dance_state_t *state) {
+    if (state->pressed) {
+        return SINGLE_HOLD;
+    }
+    if (state->count == 1) {
+        return SINGLE_TAP;
+    }
+    if (state->count == 2) {
+        return DOUBLE_TAP;
+    }
+    if (state->count == 3) {
+        return TRIPLE_TAP;
+    }
+
+    return -1;
+}
+
+// Extended Space Cadet Shift - Left ==================================
+void espc_l_finished(qk_tap_dance_state_t *state, void *user_data) {
+    espc_l_tap_state = cur_dance(state);
+    switch (espc_l_tap_state) {
+        case SINGLE_TAP:  // (
+            register_mods(MOD_LSFT);
+            register_code(KC_9);
```

Comment:
`(un)register_code16` supports shifted characters.  You can replace most of the multiple function calls with just the one function call

Eg: 
```suggestion
            register_code16(LSFT(KC_9));
```